entity-framework/core/modeling/value-conversions.md

```diff
@@ -2,145 +2,746 @@
+
+<!--
+        public class Blog
+        {
+            public string Id { get; set; }
+            public string Name { get; set; }
+
+            public ICollection<Post> Posts { get; set; }
+        }
+
+        public class Post
+        {
+            public string Id { get; set; }
+            public string Title { get; set; }
+            public string Content { get; set; }
+
+            public string BlogId { get; set; }
+            public Blog Blog { get; set; }
+        }
+-->
+[!code-csharp[CaseInsensitiveStringsModel](../../../samples/core/Modeling/ValueConversions/CaseInsensitiveStrings.cs?name=CaseInsensitiveStringsModel)]
+
+This will not work as expected if some of the `Post.BlogId` values have different casing. A value comparer can be used to correct this:
+
+<!--
+            protected override void OnModelCreating(ModelBuilder modelBuilder)
+            {
+                var comparer = new ValueComparer<string>(
+                    (l, r) => string.Equals(l, r, StringComparison.OrdinalIgnoreCase),
+                    v => v.ToUpper().GetHashCode(),
+                    v => v);
+
+                modelBuilder.Entity<Blog>()
+                    .Property(e => e.Id)
+                    .Metadata.SetValueComparer(comparer);
+
+                modelBuilder.Entity<Post>(
+                    b =>
+                        {
+                            b.Property(e => e.Id).Metadata.SetValueComparer(comparer);
+                            b.Property(e => e.BlogId).Metadata.SetValueComparer(comparer);
+                        });
+            }
+-->
+[!code-csharp[ConfigureCaseInsensitiveStrings](../../../samples/core/Modeling/ValueConversions/CaseInsensitiveStrings.cs?name=ConfigureCaseInsensitiveStrings)]
+
+> [!NOTE]
+> .NET string comparisons and database string comparisons can differ in more than just case sensitivity. This pattern works for simple ASCII keys, but may fail for keys with any kind of culture-specific characters. See [Collations and Case Sensitivity](xref:core/miscellaneous/collations-and-case-sensitivity) for more information.
+
+### Handle fixed-length database strings
+
+The previous example did not need a value converter. However, a converter can be useful for fixed-length database string types like `char(20)` or `nchar(20)`. Fixed-length strings are padded to their full length whenever a value is inserted into the database. This means that a key value of "dotnet" will be read back from the database as "dotnet              ". This will then not compare correctly with key values that are not padded.
```

Comment:
```suggestion
The previous example did not need a value converter. However, a converter can be useful for fixed-length database string types like `char(20)` or `nchar(20)`. Fixed-length strings are padded to their full length whenever a value is inserted into the database. This means that a key value of `dotnet` will be read back from the database as `dotnet              `. This will then not compare correctly with key values that are not padded.
```

Reply:
Are fences really appropriate here? These are actually string values, not, not language concepts or typpes.

Reply:
That's true, but (a) the spaces don't really show, and (b) "dotnet" may get machine-translated (to who knows what)...

Reply:
Turns out we are not allowed trailing spaces in fences, so leaving as is.

Reply:
Oh well...